ml-agents-envs/mlagents/envs/timers.py

```diff
@@ -0,0 +1,145 @@
+# # Unity ML-Agents Toolkit
+from time import perf_counter
+
+from contextlib import contextmanager
+from typing import Any, Callable, Dict, Generator, TypeVar
+
+
+class TimerNode:
+    """
+    Represents the time spent in a block of code.
+    """
+
+    __slots__ = ["children", "total", "count"]
```

Comment:
Possibly a premature optimization, but this helps reduce the memory footprint and time overhead to the overall system.

Reply:
Wow I didn't know this exists. This will be super useful once we use the profiling in a tons of places.